pkg/logql/evaluator.go

```diff
@@ -59,6 +60,10 @@ func NewLiteralParams(
 		limit:     limit,
 		shards:    shards,
 	}
+	var err error
+	p.queryExpr, err = syntax.ParseExpr(qs)
+	return p, err
```

Comment:
If we want to keep the string it might be nice to rename the function `QueryString`, can be confusing at first glance to differentiate between the logql turning `Params` into a `Query` that can be `Exec`'d when we have another function here `Query`.

Reply:
> Other than that we only call Query to get the query string for passing to error logging.

That's the sole reason to keep it around. The engine should not use the string other than for logging.

> If we want to keep the string it might be nice to rename the function QueryString,

That's a good idea.